presto-orc/src/main/java/com/facebook/presto/orc/block/ListBlockReader.java

```diff
@@ -48,20 +50,26 @@
     @Nullable
     private LongStream lengthStream;
 
-    public ListJsonReader(StreamDescriptor streamDescriptor, boolean checkForNulls, DateTimeZone hiveStorageTimeZone)
+    public ListBlockReader(StreamDescriptor streamDescriptor, boolean checkForNulls, DateTimeZone hiveStorageTimeZone)
     {
         this.streamDescriptor = checkNotNull(streamDescriptor, "stream is null");
         this.checkForNulls = checkForNulls;
 
-        elementReader = createJsonReader(streamDescriptor.getNestedStreams().get(0), true, hiveStorageTimeZone);
+        elementReader = createBlockReader(streamDescriptor.getNestedStreams().get(0), true, hiveStorageTimeZone);
+    }
+
+    public Type getType()
+    {
+        // TODO: Technically, it should be ArrayType if it is moved to spi, but most VariableWidthTypes would be fine here
+        return VARBINARY;
```

Comment:
@dain thoughts on whether it's worth threading a TypeManager through to this point to fix this TODO?